docs/x509/verification.rst

```diff
@@ -294,3 +330,82 @@ the root of trust:
             for server verification.
 
         :returns: An instance of :class:`ClientVerifier`
+
+.. class:: CustomPolicyBuilder
```

Comment:
Do we need a new `CustomPolicyBuilder`? Can they just be methods on the existing `PolicyBuilder?`

Reply:
Yeah, I also thought about that. That's definitely a possibility, but I prefer this version since it kind of follows the "make it hard to do insecure things" part of cryptography's philosophy. In this case it would be more like "make insecure things explicit", e.g. this would make using any of the "easier to get wrong" features more visible in a code review.

The first paragraph from [this comment of mine](https://github.com/pyca/cryptography/issues/11165#issuecomment-2302183750) describes my motivation pretty well too. There's also [a comment](https://github.com/pyca/cryptography/issues/11165#issuecomment-2326724920) from @woodruffw where he describes why he prefers the `CustomPolicyBuilder` API.

In the end, I would be fine with both APIs, but I have grown to like the idea of a separate `CustomPolicyBuilder`. (Although I obviously can't be fully objective since it was my idea)

Reply:
I'm not really sold on the need for a totally separate builder, but maybe we can split the difference and have the constructor be `PolicyBuilder.custom()`, that way it's not a totally separate type, but still makes clear you're doing something a bit different.

Reply:
Importantly: If they're the same class, there's a _lot_ of duplication that goes away, including in both the implementation and the tests. Then we make the focus really on the new APIs.

Reply:
If we are not adding a separate class, I don't think an implementation with `custom` would be very clean either to be honest. I think I will just put all of the new functionality into `PolicyBuilder`,  which, as you say, will make the implementation a lot more pleasant. I'll do that as the last commit in this review round so it's easily revertable in case we reach a different conclusion in the end.